docs-src/cloud/migration.md

```diff
@@ -0,0 +1,88 @@
+You can also export JSON from your previous execution history, which you can then import into your own analytics system.
+
+**Running Workflows can either be drained or migrated.**
+If your Workflow can be completed before any compelling event which drives a move to Temporal Cloud, then
+those Workflows can be automatically restarted on Temporal Cloud.
+If your Workflows need to run continuously, then you must migrate Workflows while they are running.
+To accomplish this migration, you must cancel your current Workflow and pass the current state to a new Workflow in Temporal Cloud.
+Refer to [this repository](https://github.com/temporalio/migration-example/blob/main/src/main/java/io/temporal/migration/example/README.md) for an example of migrating running Workflows in Java.
+
+If you are performing a live migration, make sure your Worker capacity can support the migration load.
+Both a [Signal](/concepts/what-is-a-signal) and a [Query](/concepts/what-is-a-query) will be executed during the course of the migration.
+Also, the Query API loads the entire history of Workflows into Workers to compute the result (if they are not already cached).
+That means that your self-hosted cluster Worker capacity will need to support having those executions in memory to serve those requests.
+The volume of these requests might be quite high to execute against all the matches to a `ListFilter`.
+
+### Considerations when resuming Workflows on a new Cluster or Namespace
+
+- **Skipping Steps:** If your Workflow steps cannot guarantee idempotency, determine if you need to skip those steps when resuming the execution in the target Namespace.
+- **Elapsed Time:** If your Workflow is “resuming sleep” when in the target Namespace, determine how you will calculate the delta for the sleep invocation in the new execution.
+- **Child Relationships:** If your Workflow has Child Workflow relationships (other than Detached Parent Close Policy children), determine how you can pass the state of those children into the parent to execute the child in a resumed state.
+- **Heartbeat state:** If you have long running activities relying on heartbeat state, determine how you can resume these activities in the target Namespace.
+- If Child Workflows are the same type as their Parent types, those will be returned in List Filters being used to gather relevant executions. Unless these are Detached `ParentClosePolicy` children, this is not what you want since the Parent/Child relationship will not be carried over to the target Namespace.
+- Long running activities that use heartbeat details will not receive the latest details in the target Namespace.
+- Duration between Awaitables inside Workflow definition needs to be considered for elapsed time accuracy when resuming in the target Namespace.
+- Signaling directly from one Workflow to another needs to handle NotFound executions in the target Namespace since they may resume out of order.
```

Comment:
Suggestion: "When Signaling directly from one Workflow to another, make sure to handle NotFound executions in the target Namespace. The Workflows may resume out of order."